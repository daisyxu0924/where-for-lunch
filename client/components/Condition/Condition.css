@import '../../styles/variables.css';

.root {
  padding: $base-spacing;
  > span, > div {
    display: inline;
  }
  > ul { 
    list-style: none;
  }

}
