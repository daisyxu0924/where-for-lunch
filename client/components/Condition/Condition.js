import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import Input from 'components/Input/Input';
import styles from './Condition.css';
import CheckboxGroup from '../CheckboxGroup/CheckboxGroup';
import foodList from '../../variables/foodList';

export default class Condition extends PureComponent {
  static propTypes = {
    condition: PropTypes.object,
    setRadius: PropTypes.func,
    changeFood: PropTypes.func,
  };

  render() {
    const { condition: { radius, foods }, setRadius, changeFood } = this.props;
    return (
      <div className={styles.root}>
        <CheckboxGroup
          title={'Foods'}
          data={{ dataSet: foodList, selected: foods }}
          changeFood={changeFood}
        />
        <div>
          <span>radius:</span>
          <Input defaultValue={radius} onBlurAction={e => setRadius(e.target.value)}></Input>
          <span>meters</span>
        </div>
      </div>
    );
  }
}

