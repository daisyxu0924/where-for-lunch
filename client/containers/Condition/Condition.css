.root {
  > span, > div {
    display: inline;
  }

  .filters {
    > div {
      display: flex;
    }
  }
}
