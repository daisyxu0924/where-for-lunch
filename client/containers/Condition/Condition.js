import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import conditionActions from 'actions/conditionActions';
import Input from 'components/Input/Input';
import Checkbox from 'components/Checkbox/Checkbox';
import { isCategoryChecked } from 'lib/utils';
import styles from './Condition.css';

class Condition extends Component {
  constructor() {
    super();
    this.choices = {
      chinese: 'Chinese',
      australian: 'Australian',
      filipino: 'Filipino',
      italian: 'Italian',
      japanese: 'Japanese',
      korean: 'Korean',
      mexican: 'Mexican',
      french: 'French',
      malaysian: 'Malaysian',
    };
  }

  handleOnChangeAction = (e) => {
    const { checked, name } = e.target;
    if (checked) {
      this.props.addCategory(name);
    } else {
      this.props.removeCategory(name);
    }
  }

  handleOnBlurAction = (e) => {
    this.props.setRadius(e.target.value);
  }

  render() {
    const { condition: { radius, categories } } = this.props;
    return (
      <div className={styles.root}>
        <span>radius:</span>
        <Input defaultValue={radius} onBlurAction={this.handleOnBlurAction} />
        <span>meters</span>
        <div className={styles.filters}>
          {Object.keys(this.choices).map(item => (
            <div key={item}>
              <label>
                <Checkbox
                  onChangeAction={this.handleOnChangeAction}
                  name={item}
                  defaultChecked={isCategoryChecked(categories, item)} />
                {this.choices[item]}
              </label>
            </div>
          ))}
        </div>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  condition: state.condition,
});

const mapDispatchToProps = dispatch =>
  bindActionCreators({
    setRadius: conditionActions.setRadius,
    addCategory: conditionActions.addCategory,
    removeCategory: conditionActions.removeCategory,
  }, dispatch);

Condition.propTypes = {
  condition: PropTypes.object,
  setRadius: PropTypes.func,
  addCategory: PropTypes.func,
  removeCategory: PropTypes.func,
};

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(Condition);
