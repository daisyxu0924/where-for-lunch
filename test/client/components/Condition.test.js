import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import Condition from 'client/components/Condition/Condition';

describe('<Condition />', () => {
  it('should render with price checkboxes.', () => {
    const condition = {};
    const component = renderer
      .create(<Condition condition={condition} />)
      .toJSON();

    expect(component).toMatchSnapshot();
  });

  it('should trigger the action when a price filter is changed.', () => {
    const condition = {};
    const action = jest.fn();
    const wrapper = mount(<Condition condition={condition} action={action} />);
    const checkbox = wrapper.find('input[type="checkbox"]').first();

    checkbox.simulate('change');

    expect(action.mock.calls.length).toBe(1);
  });
});
