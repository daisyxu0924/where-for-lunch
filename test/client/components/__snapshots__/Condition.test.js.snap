// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Condition /> should render with price checkboxes. 1`] = `
<div
  className="root"
>
  <div>
    <label>
      <input
        checked={false}
        onChange={[Function]}
        type="checkbox"
        value="1"
      />
      $
    </label>
    <label>
      <input
        checked={false}
        onChange={[Function]}
        type="checkbox"
        value="2"
      />
      $$
    </label>
    <label>
      <input
        checked={false}
        onChange={[Function]}
        type="checkbox"
        value="3"
      />
      $$$
    </label>
    <label>
      <input
        checked={false}
        onChange={[Function]}
        type="checkbox"
        value="4"
      />
      $$$$
    </label>
  </div>
  <span>
    radius:
  </span>
  <div
    className="root"
  >
    <input
      onBlur={[Function]}
      type="text"
    />
  </div>
  <span>
    meters
  </span>
</div>
`;
